src/reducer.js

```diff
@@ -2,11 +2,16 @@
 import type { Action } from 'redux';
 import type { Location } from 'history';
 import { LOCATION_CHANGED } from './action-types';
+import isEqual from 'lodash.isequal';
 
 export default (state: ?Location | Object = {}, action: Action) => {
   if (action.type === LOCATION_CHANGED) {
     // No-op the initial route action
-    if (state && state.pathname === action.payload.pathname) {
+    if (
+      state &&
+      state.pathname === action.payload.pathname &&
+      isEqual(state.query, action.payload.query)
```

Comment:
Ah yes :) that's a perfect solution for me, but note that:

```js
'?a=a&b=b' !== '?b=b&a=a'
```
while
```js
_.isEqual({a: 'a', b: 'b'}, {b: 'b', a: 'a'}) === true
```

That said, I think what you are actually looking for here is "did the url change" which is correctly reported in the `search` check. I'll change the PR now.